### ISSUE TITLE
[Bug]: sampler warmup misses the plain-greedy path — spec-decode rejection kernels JIT on the first request

### ISSUE THREAD

**Issue by TSUMUGI-XE** [vllm-project/vllm#54455]
Sibling of #54425 — same function, a second missing dimension. Filing separately because the
missing path and the observable failure are different; I have not touched that issue.

`SamplingParams.for_sampler_warmup()` is documented as "Set parameters to exercise all sampler
logic", but every warmup request gets `temperature=0.9`, so the warmup never takes the
`needs_logits_processing == False` early return in `Sampler.apply_sampling_params()` — the path
a plain `temperature=0` request uses.

With spec decoding that has a visible consequence. The startup warmup *does* reach
`rejection_sample()` (`_run_decode_step` populates `scheduled_spec_decode_tokens`), but it only
ever arrives there in fp32. dtype is a Triton compile key, so
`_compute_local_logits_stats_kernel`, `_rejection_kernel` and `_resample_kernel` are never
compiled for the model dtype and JIT on the first real greedy request:

```
startup (for_sampler_warmup) : needs_logits_processing=True  -> fp32
first temperature=0 request  : needs_logits_processing=False -> bfloat16   <- JIT here
```

`num_logits` is the same on both sides, so shape is held constant; warming the model dtype
alone takes the three `jit_monitor` warnings to zero. Measured first-request penalty on my box:
**+613 ms (52.1%)**. With `--jit-monitor-mode=error` the monitor is configured to raise on that
request (`vllm/utils/jit_monitor.py`), so this is not only a latency issue.

### The general shape (why this and #54425 are the same bug)

`warmup()` enumerates the **batch-shape** space carefully — `decode_steps`
(`vllm/v1/worker/gpu/warmup.py` L391-409 on main `@cacc429f6`) covers all-spec, mixed
spec/non-spec, single-request and no-draft-token variants, and L222-225 budgets for exactly
that (`num_decode_steps = 5 if num_spec_steps > 0 else 3`). But it samples the
**sampling-parameter** space at exactly one point — L280:

```python
sampling_params = SamplingParams.for_sampler_warmup()   # one object, all requests
```

#54425 is one axis of that point (no `seed` → FlashInfer is chosen, the seeded native
top-k/top-p + Gumbel path never runs). This report is another (`temperature=0.9` →
`needs_logits_processing` short-circuit never runs → the greedy dtype is never compiled).
Both are "the docstring says all sampler logic, and one dimension is pinned".

### Suggested fix

Same shape as #54425's: **run more than one sampler warmup configuration**, rather than
special-casing either axis. Concretely, either

1. add a second pass with a plain `SamplingParams()` (greedy) alongside the existing one, or
2. have `for_sampler_warmup()` return a small *list* of configurations and let `warmup()`
   iterate over it the way it already iterates over `decode_steps` (L409) — which would
   close #54425 and this one with the same change, and rides the structure already there.

I have a working local patch for the narrow greedy case (18 lines in
`vllm/v1/worker/gpu/warmup.py`) that takes the JIT warnings to zero, but it clears an internal
sampler buffer for the replay, which I don't think is the right shape to upstream. Happy to
send whichever of (1)/(2) you prefer as a PR.

### Environment

Intel XPU (Arc Pro B70), vLLM main `@cacc429f6` (2026-08-28), spec decode `qwen3_5_mtp`,
Qwen3.6-35B-A3B MXFP4. I carry one unrelated local patch (an mxfp4 `register_fake`);
`sampling_params.py` and `warmup.py` are stock. No CUDA box here to confirm, and as far as I
can tell nothing in this path branches on platform, but I can't verify that claim on hardware.

**Comment by bandham-manikanta** [comment]
Hi @TSUMUGI-XE, are you actively working on a PR for this? If not, I'm happy to take this on alongside #54425 to support multi-configuration sampler warmup across greedy, seeded native, and FlashInfer paths.

**Comment by TSUMUGI-XE** [comment]
Not working on a PR, so please go ahead. Covering both axes in one
multi-configuration warmup is the right shape — better than the single-axis
fix I had sketched.

What I can offer instead is verification. I have the reproduction here:
Intel XPU, spec decode `qwen3_5_mtp`, where the first plain-greedy request
currently JITs the three rejection-sampling kernels (`jit_monitor` reports 3,
~+613 ms). Ping me when you have a branch and I'll run it and report the
counts before and after.

One implementation detail worth keeping: the greedy pass has to go through the
spec-decode decode step, not just prefill — otherwise it never reaches
`rejection_sample()` and the kernels stay uncompiled.

**Comment by bandham-manikanta** [comment]
Opened #54630 with the multi-config warmup. @TSUMUGI-XE please take a look and let me know how it tests on your Intel XPU setup!
